src/modules.js

```diff
@@ -447,7 +447,7 @@ function exportRuntime() {
       }
     }
 
-    if (unexported.length) {
+    if (!MINIMAL_RUNTIME && unexported.length) {
```

Comment:
This block runs inside `ASSERTIONS` builds, i.e. it is intended to help at debug time to catch errors. So it would be ok to enable that for `MINIMAL_RUNTIME` as well to help debug issues there. Or was there some kind of issue emitting this code into `MINIMAL_RUNTIME` builds?

Reply:
The issue is that when we start exporting the symbols, the export is done after this getter is installed, so:

```js
// Install getter here.
...
// Export on `Module`
Module[_my_exported_function] = ...; // Throws an error
```

In general as we mentioned throughout this PR, there are issues considering what's exported for usage inside the module, and what's exported to the outside world. So we could either find a way to fixup this `unexported` list, or install the assertion later in the process when using `MINIMAL_RUNTIME`.

Reply:
Good point, added a comment 👌